PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java

```diff
@@ -134,20 +143,19 @@ public AddNewClusterForm(@NotNull final Project project, @Nullable RefreshableNo
             CardLayout layout = (CardLayout) (clusterCardsPanel.getLayout());
             layout.show(clusterCardsPanel, (String) e.getItem());
 
-            // if "HDInsight Cluster" is chose, "Basic Authentication" should be the only one authentication method
+            // if "HDInsight Cluster" is chose, "No Authentication" should not exist
             if (isHDInsightClusterSelected()) {
-                authComboBox.setSelectedItem(authComboBox.getModel().getElementAt(0));
-                authComboBox.setEnabled(false);
+                authComboBox.setModel(new ImmutableComboBoxModel(AuthTypeOptions.HDICluster.getOptionTypes()));
```

Comment:
Good design! It's very clear now! One more suggestion is to use 2 variables to hold two `ImmutableComboBoxModel`. Therfore, the previously input credential can be recorded whenever user switch back to the current cluster type.